README.md

```diff
@@ -4,133 +4,99 @@ Beancounter
-The tool supports various types of wallets, including multisig + segwit.
+Beancounter is a command line utility to audit the balance of Hierarchical Deterministic (HD) wallets at a given point in time (or block height). The tool is designed to scale and work for wallets with a large number of addresses or a large number of transactions.
 
-We picked Go as the programming language, because Go makes building the tool for different platforms trivial. We also
-appreciate the ability to distribute static binaries which don't have external dependencies.
+The tool supports various types of wallets ranging from simple watch wallets to more complicated multisig + segwit.
 
-We built Beancounter because we wanted a tool which can compute the balance of a wallet at any
-given block height.
+Beancounter currently supports two types of backends to query the blockchain:
+1. Electrum public servers. When using these servers, Beancounter behaves in a similar fashion to an Electrum client wallet. The servers are queried for transaction history for specific addresses. Using Electrum servers is easiest but requires trusting public servers to return accurate information. There is also potential privacy exposure.
 
-The process to determine a deterministic wallet's balance involves the following:
-1. Use the blockchain to build an address index. Bitcoin Core doesn't offer this, but Btcd,
-   Electrum, and a few other tools do.
-2. Derive external (receive) and internal (change) addresses.
-3. Fetch the transaction history for a single address. Prune the history at the desired block height.
-4. Compute the credit and debit for each transaction. Repeat until we find a large number of
-   unused addresses.
-
-This tool currently supports two backends: Electrum and Btcd. Using Electrum is easier (connects to
-public nodes) but has some disadvantages (need to trust a third party, potential privacy leak, etc.).
-Using Btcd requires running a private node and the initial sync can take a long time.
+2. Private Btcd node. Btcd is a Bitcoin full node which implements transaction indexes. Setting up a Btcd node can take some time (the initial sync takes ~7 days) and requires maintaining the node up-to-date. The benefit is however a higher level of guarantee that the transaction history is accurate.
 
 ![logo](https://raw.githubusercontent.com/square/beancounter/master/coffee.jpg)
 
-Getting Started (using Electrum)
-================================
+Getting Started
+===============
 
-```
-brew install dep
+Installing
+----------
+If missing, install [https://github.com/golang/dep](dep)
 
-dep ensure
-go run main.go -m 2 -n 4 --account 1234
+```
+$ git clone https://github.com/square/beancounter/
+$ cd beancounter
+$ dep ensure
+$ go build
 ```
 
-Sample Output
-=============
+Deriving the child pubkey
+-------------------------
+Let's imagine we want to track the balance of `tpubD8L6UhrL8ML9...` and the derivation being used is `m/1'/1234/change/index`.
```

Comment:
what would be the benefit of deriving the child pubkey separately from the address derivation?

Reply:
Most people won't need to derive child pubkeys. It makes all the other commands simpler.